packages/x-telemetry/src/sender.ts

```diff
@@ -0,0 +1,65 @@
+import { ponyfillGlobal } from '@mui/utils';
+import { getMuiXTelemetryEnv } from './config';
+import telemetryContext from './context';
+import { TelemetryEvent } from './types';
+
+function shouldSendTelemetry(): boolean {
+  // Disable collection of the telemetry
+  // in non-production environment
+  if (process.env.NODE_ENV === 'production') {
+    return false;
+  }
+
+  // Priority to the config (e.g. in code, env)
+  const envIsCollecting = getMuiXTelemetryEnv('IS_COLLECTING');
+  if (typeof envIsCollecting === 'boolean') {
+    return envIsCollecting;
+  }
+
+  if (typeof telemetryContext.config.isCollecting === 'boolean') {
+    return telemetryContext.config.isCollecting;
+  }
+
+  // Disabled by default
+  return false;
+}
+
+function getLicenseKey(): string | undefined {
+  // eslint-disable-next-line no-underscore-dangle
+  return ponyfillGlobal.__MUI_LICENSE_INFO__?.key || undefined;
+}
+
+function sendMuiXTelemetryEvent(event: TelemetryEvent | null) {
+  if (!event || !shouldSendTelemetry()) {
+    return;
+  }
+
+  const eventPayload = {
+    ...event,
+    context: {
+      ...telemetryContext.traits,
+      licenseKey: getLicenseKey(),
```

Comment:
We should clarify what "anonymized" means in the context of the data that are being sent.
I'm not sure how the license key fits this definition. Any thoughts?

Reply:
Sure, it should be completely anonymous, I've updated the code to use license key hash instead of raw license key to keep it completely anonymized

Reply:
Strictly speaking, is there something preventing us from storing the license key we generate and then use them to do a lookup to match the hashed license key to its clear version?

I know we won't do it, but people could be worried because we can do it

Reply:
That's a good point. We need to have clear terms and likely separate the calls (and separate opt-ins)